### PR TITLE
Add support for user settings to CMake.

### PR DESCRIPTION
**Pull request by haydenroche5**

### CODE REVIEW

**Review by dgarske — CHANGES_REQUESTED** [review]
@haydenroche5 , thanks for adding support for this. Can you also update the version number in this file to v4.6.0? I've made sure that release instructions have this noted.

**Comment by haydenroche5** [comment]
> @haydenroche5 , thanks for adding support for this. Can you also update the version number in this file to v4.6.0? I've made sure that release instructions have this noted.

Sure, added.

**Review by dgarske — APPROVED** [review]

**Comment by haydenroche5** [comment]
I added an option to disable generation of config.h and the defining of HAVE_CONFIG_H. See https://www.wolfssl.com/forums/post5684.html#p5684.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1292,11 +1299,6 @@ file(APPEND ${CYASSL_OPTION_FILE} " */\n")
 file(READ ${OPTION_FILE} OPTION_FILE_CONTENTS)
 file(APPEND ${CYASSL_OPTION_FILE} ${OPTION_FILE_CONTENTS})
 
-add_definitions(${WOLFSSL_DEFINITIONS})
```
Comment (haydenroche5) [inline CMakeLists.txt:1295]: This was duplicate code.

**Comment by haydenroche5** [comment]
retest this please

**Pull request merged**